tidy3d/components/monitor.py

```diff
@@ -158,6 +158,32 @@ class AbstractFieldMonitor(Monitor, ABC):
         description="Collection of field components to store in the monitor.",
     )
 
+    interval_space: Tuple[
+        pydantic.PositiveInt, pydantic.PositiveInt, pydantic.PositiveInt
+    ] = pydantic.Field(
+        (1, 1, 1),
+        title="Spatial interval",
+        description="Number of grid step intervals between monitor recordings. If equal to `1`, "
+        "there will be no downsampling. If greater than 1, fields will be downsampled "
+        "and automatically colocated.",
+    )
+
+    colocate: bool = pydantic.Field(
+        None,
+        title="Colocate fields",
+        description="Toggle whether fields should be colocated to grid cell centers. Default: "
+        "`False` if `interval_space` is 1 in each direction, `True` if `interval_space` is "
+        "greater than one in any direction.",
+    )
+
+    @pydantic.validator("colocate", always=True)
```

Comment:
This also needs to be fixed @tylerflex

Reply:
Thanks, added to https://github.com/flexcompute/tidy3d/issues/392 and feel free to add others if you find them.